.github/workflows/build.yml

```diff
@@ -0,0 +1,106 @@
+name: Build and Test ldsCtrlEst
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Cache vcpkg binaries
+      uses: actions/cache@v3
+      with:
+        path: ${{ runner.os == 'Windows' && 'C:\\Users\\runneradmin\\AppData\\Local\\vcpkg\\' || '~/.cache/vcpkg' }}
+        key: vcpkg-binary-${{ matrix.os }}-${{ hashFiles('vcpkg.json') }}
+        restore-keys: |
+          vcpkg-binary-${{ matrix.os }}-
+    - name: Cache vcpkg directory
+      uses: actions/cache@v3
+      with:
+        path: vcpkg
+        key: vcpkg-${{ matrix.os }}-${{ hashFiles('vcpkg.json') }}
+        restore-keys: |
+          vcpkg-${{ matrix.os }}-
+    - name: Set up vcpkg for Windows
+      if: runner.os == 'Windows'
+      shell: pwsh
+      run: |
+        if (Test-Path -Path "vcpkg") {
+          Remove-Item -Recurse -Force "vcpkg"
+        }
+        git clone https://github.com/microsoft/vcpkg.git
+        cd vcpkg
+        .\bootstrap-vcpkg.bat
+        .\vcpkg integrate install
+    - name: Set up vcpkg for macOS/Linux
+      if: runner.os != 'Windows'
+      run: |
+        rm -rf vcpkg
+        git clone https://github.com/microsoft/vcpkg.git
+        cd vcpkg
+        ./bootstrap-vcpkg.sh
+        ./vcpkg integrate install
```

Comment:
All these vcpkg steps can likely be taken care of for you by someone else:
- https://github.com/marketplace/actions/run-vcpkg
- https://github.com/marketplace/actions/run-cmake